src/main/java/com/google/sps/servlets/EventsServlet.java

```diff
@@ -0,0 +1,133 @@
+
+import com.google.api.client.util.DateTime;
+import com.google.api.services.calendar.Calendar;
+import com.google.api.services.calendar.model.Event;
+import com.google.api.services.calendar.model.Events;
+import com.google.api.services.calendar.model.EventDateTime;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import com.google.sps.utils.ServletUtils;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet for obtaining the Calendar ID of a group and adding or obtaining events
+ */
+@WebServlet("/events")
+public class EventsServlet extends AbstractEventsServlet {
+  private final int MAX_EVENTS = 50;
+
+  /**
+   * Obtain a list of events from the specified group if given the Calendar ID.
+   * Obtain the Calendar ID of a group given the Group ID. If both provided, return list of events.
+   * These values should be in the query string.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String calendarId = getParameter(request, GROUP_CALENDARID_PROPERTY);
+    String groupId = getParameter(request, GROUP_ID_PROPERTY);
+
+    if (!Strings.isNullOrEmpty(calendarId)) {
+      try {
+        Events events = getEventsList(getCalendarService(), calendarId); 
+
+        response.setContentType("application/json;");
+        response.getWriter().println(ServletUtils.gson.toJson(events));
+      } catch (Exception authenticationError) {
+        authenticationError.printStackTrace();
+        response.getWriter().println("Invalid credentials or calendar ID");
+      }
+    } else if (!Strings.isNullOrEmpty(groupId)) {
+      try {
+        calendarId = ServletUtils.getGroupProperty(groupId, GROUP_CALENDARID_PROPERTY);
+
+        response.setContentType("text/plain");
+        response.getWriter().println(calendarId);
+      } catch (Exception entityError) {
+        entityError.printStackTrace();
+        response.getWriter().println("Invalid group ID");
+      }
+    } else {
+      ServletUtils.printBadRequestError(response, EVENTS_GET_BAD_REQUEST_MESSAGE);
+    }
+  }
+
+  /**
+   * Create an event and add it to the calendar of the specified group. Query string should contain groupId
+   * POST body should contain: title, location (opt), description (opt), start time, and end time.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String groupId = getParameter(request, GROUP_ID_PROPERTY);
+    String eventTitle = getParameter(request, "title");
+    String eventStart = getParameter(request, "start");
+    String eventEnd = getParameter(request, "end");
+
+    if(!Strings.isNullOrEmpty(groupId) && !Strings.isNullOrEmpty(eventTitle) 
+        && !Strings.isNullOrEmpty(eventStart) && !Strings.isNullOrEmpty(eventEnd)) {
```

Comment:
Which do you think looks cleaner (I'm not sure myself): 
if(!(Strings.isNullOrEmpty(groupId) || Strings.isNullOrEmpty(eventTitle) 
        || Strings.isNullOrEmpty(eventStart) || Strings.isNullOrEmpty(eventEnd))) {

I just used DeMorgan's here, but I know java gets a little picky with some OR statements.

Reply:
i think the way i have it makes it more clear that it requires all the parameters but yours is definitely cleaner so not sure which is better :/ we could ask a third person what they think